### PR TITLE
[FIX] Use NumPy in _fit and _predict_naive to resolve indexing issue

### LINKED ISSUE #4460: [BUG] `NaiveForecaster` performance issue when forecasting in-sample
**Describe the bug**
I want to detrend before forecasting by normalizing my inputs to the latest value or to the mean within a window. It seems that I should be able to use `Detrender(forecaster=NaiveForecaster())` in a pipeline to achieve this. The performance makes it impossible. Using the default polynomial forecaster to detrend is much faster than using a constant forecaster.

**To Reproduce**
```python
from sktime.pipeline import make_pipeline
from sktime.forecasting.compose import make_reduction
from sktime.forecasting.naive import NaiveForecaster
from sktime.transformations.series.detrend import Detrender

import numpy as np
import pandas as pd
import xgboost as xgb

forecaster = make_pipeline(
    Detrender(),
    make_reduction(xgb.XGBRegressor())
)

y_train = pd.Series(np.random.rand(500), index=pd.date_range('1900-01-01', periods=500))
forecaster.fit(y_train) # takes ~0.4s

forecaster = make_pipeline(
    Detrender(forecaster=NaiveForecaster()),
    make_reduction(xgb.XGBRegressor())
)

forecaster.fit(y_train) # takes ~12s
```

**Expected behavior**
`Detrender(forecaster=NaiveForecaster())` should be as fast or faster than the default `Detrender(forecaster=PolynomialTrendForecaster(degree=1))`.

**Context**
MacBook Pro M1

**Versions**
<details>
System:
    python: 3.10.10 (main, Feb 16 2023, 02:46:59) [Clang 14.0.0 (clang-1400.0.29.202)]
executable: /Users/kgeis/temp/mystuff/sp_prob/bin/python
   machine: macOS-12.6.3-arm64-arm-64bit

Python dependencies:
          pip: 23.0.1
       sktime: 0.17.1
      sklearn: 1.2.2
        numpy: 1.23.5
        scipy: 1.10.1
       pandas: 1.5.3
   matplotlib: 3.7.1
       joblib: 1.2.0
  statsmodels: 0.13.5
        numba: 0.56.4
     pmdarima: 2.0.3
      tsfresh: 0.20.0
   tensorflow: None
tensorflow_probability: None
</details>

### PR DESCRIPTION
**Pull request by Arpithajain26**
Fixes #4460

What does this implement/fix? Explain your changes.

This PR updates the NaiveForecaster implementation to handle indexing issues when broadcasting to panel/hierarchical data.

Key changes:

Converted target series (y) to NumPy arrays inside _fit for consistent handling and improved performance.

Updated _predict_naive to use NumPy instead of pandas indexing.

Ensured outputs are converted back to pandas with the correct index to maintain API consistency.

This resolves the slow broadcasting problem at large scales and avoids indexing mismatches.

Does your contribution introduce a new dependency?

No, it does not introduce any new dependencies.

What should a reviewer concentrate their feedback on?

Correctness of NumPy conversion in _fit and _predict_naive.

Preservation of the expected pandas index in returned forecasts.

Performance improvements when handling panel/hierarchical data.

Did you add any tests for the change?

No new tests were added. The existing test suite should already cover this functionality, but I am happy to extend test cases if needed.

Any other comments?

This is my first contribution to the project.

I took some help from an AI tool (ChatGPT) to draft the code and PR description, and I’ve reviewed the changes myself.

Feedback is very welcome to help me learn and contribute better in the future.

### CODE REVIEW

**Comment by Arpithajain26** [comment]
“Hi @fkiraly, just wanted to kindly ask if you could take a look at this PR when you have some time. Please let me know if any changes are needed. Thank you!
